### PR TITLE
CEDS-974 Previous documents - multiplicity

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/controllers/supplementary/PackageInformationController.scala [hunk 1]
```diff
@@ -68,23 +67,26 @@ class PackageInformationController @Inject()(
 
   def remove(
     id: Option[String]
-  )(implicit authRequest: AuthenticatedRequest[AnyContent], packagings: Seq[PackageInformation]) =
+  )(implicit authRequest: AuthenticatedRequest[AnyContent], packages: Seq[PackageInformation]): Future[Result] =
```
Comment (PauloPortugal) [inline app/controllers/supplementary/PackageInformationController.scala]: :+1: `packages` instead `packagings`

File: app/controllers/supplementary/PreviousDocumentsController.scala [hunk 2]
```diff
@@ -32,30 +34,64 @@ import views.html.supplementary.previous_documents
 import scala.concurrent.{ExecutionContext, Future}
 
 class PreviousDocumentsController @Inject()(
-  appConfig: AppConfig,
   override val messagesApi: MessagesApi,
   authenticate: AuthAction,
+  errorHandler: ErrorHandler,
   customsCacheService: CustomsCacheService
-)(implicit ec: ExecutionContext)
+)(implicit ec: ExecutionContext, appConfig: AppConfig)
```
Comment (PauloPortugal) [inline app/controllers/supplementary/PreviousDocumentsController.scala]: :+1:

File: app/controllers/supplementary/PreviousDocumentsController.scala [hunk 3]
```diff
@@ -32,30 +34,64 @@ import views.html.supplementary.previous_documents
...
+
+    val boundForm = form.bindFromRequest()
+
+    val actionTypeOpt = request.body.asFormUrlEncoded.map(FormAction.fromUrlEncoded(_))
+
+    val cachedData = customsCacheService
+      .fetchAndGetEntry[PreviousDocumentsData](supplementaryCacheId, formId)
+      .map(_.getOrElse(PreviousDocumentsData(Seq.empty)))
+
+    cachedData.flatMap { cache =>
```
Comment (PauloPortugal) [inline app/controllers/supplementary/PreviousDocumentsController.scala]: In the name of `encapsulation` and readability, I would extract the `Add`, `Remove` and `Continue` implementation to it's own methods.
Not a blocker ;)

Comment (PauloPortugal) [reply app/controllers/supplementary/PreviousDocumentsController.scala]: I still :heart: this approach! :)

File: app/views/supplementary/previous_documents.scala.html [hunk 4]
```diff
@@ -21,7 +21,7 @@
 @import uk.gov.hmrc.play.views.html._
 @import utils.RadioOption
 
-@(appConfig: AppConfig, form: Form[Document])(implicit request: Request[_], messages: Messages)
+@(form: Form[Document], documents: Seq[Document])(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
```
Comment (PauloPortugal) [inline app/views/supplementary/previous_documents.scala.html]: Coolio! I am going to do the same thing on mine!

File: conf/messages.en [hunk 5]
```diff
@@ -511,8 +516,8 @@ supplementary.transactionType.identifier.error = Further information on this tra
 
 supplementary.packageInformation.title = Package Information
 supplementary.packageInformation.remove = Remove Packaging Information
-supplementary.packageInformation.table.heading=1 Package added
-supplementary.packageInformation.table.multiple.heading={0} Packages added
+supplementary.packageInformation.table.heading= 1 Package added
```
Comment (PauloPortugal) [inline conf/messages.en]: :thinking: Interesting!

Comment (PatrykRudnicki) [reply conf/messages.en]: Only spaces after =

**Pull request merged**